perfmetrics/scripts/generate_files.py

```diff
@@ -41,29 +61,46 @@ def generate_files_and_upload_to_gcs_bucket(destination_blob_name, num_of_files,
           out.truncate(1024 * 1024 * int(file_size))
         if(file_size_unit.lower() == 'kb'):
           out.truncate(1024 * int(file_size))
+        if(file_size_unit.lower() == 'b'):
+          out.truncate(int(file_size))
+
+    num_files = os.listdir(TEMPORARY_DIRECTORY)
+    if not num_files:
+      return 0
 
     # Uploading batch files to GCS
-    process = Popen(
-        'gsutil -m cp -r {}/* {}'.format(TEMPORARY_DIRECTORY,
-                                         destination_blob_name),
-        shell=True)
-    process.communicate()
-    exit_code = process.wait()
-    if(exit_code != 0):
-      print('Exited with code {}'.format(exit_code))
-      subprocess.call('bash', shell=True)
+    if upload_to_gcs_bucket:
+      process = Popen(
+          'gsutil -m cp -r {}/* {}'.format(TEMPORARY_DIRECTORY,
+                                          destination_blob_name),
+          shell=True)
+      process.communicate()
+
+      exit_code = process.wait()
+      if(exit_code != 0):
+        if logging_method == 'output_file':
+          print('Exited with code {}'.format(exit_code))
+          subprocess.call('bash', shell=True)
+        elif logging_method == 'return':
+          return exit_code
 
     # Copying batch files from temporary to local destination folder:
     subprocess.call(
         'cp -r {}/* {}'.format(TEMPORARY_DIRECTORY, local_destination_folder),
         shell=True)
 
     # Deleting batch files from temporary folder:
-    subprocess.call('rm -r {}/*'.format(TEMPORARY_DIRECTORY), shell=True)
+    subprocess.call('rm -rf {}/*'.format(TEMPORARY_DIRECTORY), shell=True)
 
     # Writing number of files uploaded to output file after every batch uploads:
-    logmessage('{}/{} files uploaded to {}\n'.format(file_num, num_of_files,
-                                                     destination_blob_name))
+    message = '{}/{} files created.\n'.format(min(file_num, num_of_files), num_of_files)
+    logmessage(message, logging_method)
+
+    if file_num > num_of_files:
+      return 0
```

Comment:
why do you need this check? in what scenario does it happen

Reply:
since we are iterating till num_of_files + BATCH_SIZE + 1 to be on the safe side, which is extra than what we want. So we have to break the loop manually.

Reply:
removed.